ocs_ci/ocs/cephfs_workload.py

```diff
@@ -0,0 +1,214 @@
+            self.project,
+            self.tmp_path,
+        )
+        # deploy the workload, starting the log reader/writer pods
+        logger.info(
+            "starting log reader/writer workload via Deployment, one pod per worker"
+        )
+        self.workload_file.create()
+
+        logger.info("waiting for all pods of the workload Deployment to run")
+        self.ocp_pod = ocp.OCP(kind="Pod", namespace=self.project.namespace)
+        try:
+            self.ocp_pod.wait_for_resource(
+                resource_count=self.deploy_dict["spec"]["replicas"],
+                condition=constants.STATUS_RUNNING,
+                error_condition=constants.STATUS_ERROR,
+                timeout=300,
+                sleep=30,
+            )
+        except Exception as ex:
+            # this is not a problem with feature under test, but with infra,
+            # cluster configuration or unrelated bug which must have happened
+            # before this test case
+            error_msg = "unexpected problem with start of the workload, cluster is either misconfigured or broken"
+            logger.exception(error_msg)
+            logger.debug(self.workload_file.describe())
+            raise exceptions.UnexpectedBehaviour(error_msg) from ex
+
+    def fetch_and_validate_data(self):
+        """
+        While the workload is running, try to validate the data
+        from the cephfs volume of the workload.
+
+        Raise:
+            NotFoundError: When the given volume is not found in given spec
+            Exception: When the data verification job failed
+
+        """
+        # if no obvious problem was detected, run the logreader job to validate
+        # checksums in the log files (so that we are 100% sure that nothing went
+        # wrong with the IO or the data)
+        with open(constants.LOGWRITER_CEPHFS_READER, "r") as job_file:
+            job_dict = yaml.safe_load(job_file.read())
+        # if we are running in disconnected environment, we need to mirror the
+        # container image first, and then use the mirror instead of the original
+        if config.DEPLOYMENT.get("disconnected"):
+            update_container_with_mirrored_image(self.deploy_dict["spec"]["template"])
+        # drop topology spread constraints related to zones
+        topology.drop_topology_constraint(
+            job_dict["spec"]["template"]["spec"], topology.ZONE_LABEL
+        )
+        # we need to match number of jobs with the number used in the workload
+        job_dict["spec"]["completions"] = self.deploy_dict["spec"]["replicas"]
+        job_dict["spec"]["parallelism"] = self.deploy_dict["spec"]["replicas"]
+        # and reffer to the correct pvc name
+        try:
+            link_spec_volume(
+                job_dict["spec"]["template"]["spec"],
+                "logwriter-cephfs-volume",
+                self.pvc_dict["metadata"]["name"],
+            )
+        except Exception as ex:
```

Comment:
the exception is too wide. Please catch the specific one/s

Reply:
Fixed

Reply:
`Exception` is still here

Reply:
Okay, I removed it.